linbox/solutions/smith-form.h

```diff
@@ -56,22 +55,22 @@ namespace LinBox
         for( auto dit : SmithDiagonal ) {
             if (R.areEqual(dit,si)) ++num;
             else {
-                c.push_back(EC(Ring::Element)(si,num));                
+                c.push_back(SmithPair<Ring>(si,num));
```

Comment:
Isn't it a case where all `push_back`s should be replaced by `emplace_back`s ?

Reply:
yes, done.